docs/rules.rst

```diff
@@ -0,0 +1,622 @@
+Please check the individual template description below.
+
+
+Rule templates
+--------------
+
+There exist a number of predefined rule templates for generating firewall
+rules through ferm. Each rule definition is referencing the used template
+through its ``type`` key. The templates are located in the
+``templates/etc/ferm/ferm.d/`` directory.
+
+Following a list of the available rule templates which can be used to
+create custom rules:
+
+* ``accept``: Template to create rules that match interfaces, ports, remote
+  IP addresses/subnets and can accept the packets, reject, or redirect to
+  a different chain. The following template-specific YAML keys are supported:
+
+    ``accept_any``
+      Optional. ?
+
+    ``daddr``
+      Optional. List of destination IP addresses or networks to which the
+      rule is applied.
+
+    ``dport``
+      Optional. List of destination ports to which the rule is applied.
+
+    ``enabled``
+      Optional. Enable rule definition. Possible values: ``True`` or
+      ``False``. Defaults to ``True``.
+
+    ``include``
+      Optional. Custom ferm configuration file to include.
+
+    ``interface``
+      Optional. List of network interfaces for incoming packets to which
+      the rule is applied.
+
+    ``interface_present``
+      Optional. Same as ``item.interface`` but first check if specified
+      network interfaces exists before adding the firewall rules.
+
+    ``multiport``
+      Optional. Use ``iptables multiport`` extension. Possible values:
+      ``True`` or ``False``. Defaults to ``False``.
+
+    ``name``
+      Optional. Set rule name in places where it is referenced.
+
+    ``outerface``
+      Optional. List of network interfaces for outgoing packets to which
+      the rule is applied.
+
+    ``outerface_present``
+      Optional. Same as ``item.outerface`` but first check if specified
+      network interface exists before adding the firewall rule.
+
+    ``protocol``
+      Optional. Network protocol to which the rule is applied.
+
+    ``protocol_syn``
+      Optional. Match TCP packet with only the SYN flag set. Possible
+      values ``True`` or ``False``. If set to ``False`` it will match all
+      other packets except the ones with only the SYN flag set. Defaults
+      to unset.
+
+    ``realgoto``
+      Optional. ?
+
+    ``reject_with``
+      Optional. Define reject message being sent when the rule ``item.target``
+      is set to ``REJECT``. Defaults to ``icmp-admin-prohibited``.
+
+    ``saddr``
+      Optional. List of source IP addresses or networks to which this rule is
+      applied.
+
+    ``sport``
+      Optional. List fo source ports to which the rule is applied.
+
+    ``state``
+      Optional. Connection state which should be matched. Possible values:
+      ``INVALID``, ``ESTABLISHED``, ``NEW``, ``RELATED``, ``UNTRACKED`` or
+      comma-separated combination thereof.
+
+    ``subchain``
+      Optional. ?
```

Comment:
This is a `ferm` keyword which allows easier definition of custom chains. http://ferm.foo-projects.org/download/2.1/ferm.html#location_keywords In `debops.ferm` this variable exists so that some rules can be split into separate chains depending on number of resulting `iptables` entries, for example if a rule results in more than 3 entries, it will automatically be split into a custom chain.